deploy/init-db/init-db.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -e
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# Array of usernames
+RESOURCE_PROVIDERS=("ucp" "applications_rp")
+
+# Create databases and users
+for RESOURCE_PROVIDER in "${RESOURCE_PROVIDERS[@]}"; do
+    echo "Creating database and user for $RESOURCE_PROVIDER"
+    psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" <<-EOSQL
+        CREATE USER $RESOURCE_PROVIDER WITH PASSWORD '$POSTGRES_PASSWORD';
+        CREATE DATABASE $RESOURCE_PROVIDER;
+        GRANT ALL PRIVILEGES ON DATABASE $RESOURCE_PROVIDER TO $RESOURCE_PROVIDER;
+EOSQL
+done
+
+# Create tables within those databases
+for RESOURCE_PROVIDER in "${RESOURCE_PROVIDERS[@]}"; do
```

Comment:
What if we have resource provider as a column in the **resources** table? Would any of the resources created by these two separate resource providers have the same resource id by any chance? This would make adding a custom resource provider easier in my opinion.

1. **resource_providers** table.
2. **resources** table with an additional column of **resource_provider**.
3. Adding a new **resource provider** would just be an addition to the **resource_providers** table.

Otherwise, are we going to create a new database for each resource provider?

Reply:
> Would any of the resources created by these two separate resource providers have the same resource id by any chance?

No this cannot happen. 

I don't see the advantage of an extra column. We actually want an RBAC boundary so that RP's can't reach each-other's data. 

> Otherwise, are we going to create a new database for each resource provider?

Yes. Right now we have 3, soon we'll have 4. I think that's reasonable.

Reply:
So, if a Radius user creates a new resource provider, will that create a new DB schema?

Reply:
I am talking about the user defined resource providers and resource types.